### PR TITLE
Fix color picker bug when multiple colorpickers are present

### LINKED ISSUE #25013: Same background for multiple colorpickers
#### Describe the bug

Multiple color pickers will have the same background color even though they have different color values

#### Expected behavior

Each input must have a different background color
2021-06-18 at 1 21 AM

Actual

2021-06-18 at 1 21 AM

#### Steps to Reproduce

You will need a PrestaShop backoffice page which has multiple color pickers, I'm not aware of any page having that.
I discovered the bug on my module page

#### Additional information

* PrestaShop version: 1.7.8.0

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Fix color picker bug when multiple colorpickers are present<br>Duplicate of #25012 by @unlocomqx (:heart_eyes_cat: to this contributor)
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25013
| How to test?      | Check #25012 (tested by @hibatallahAouadni) & #25013

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27114)

### CODE REVIEW

File: js/jquery/plugins/jquery.colorpicker.js [hunk 1]
```diff
@@ -30,6 +30,7 @@
   var $o;
 
   $.fn.mColorPicker = function(options) {
+    var inputs = $(this);
```
Comment (PierreRambaud) [inline js/jquery/plugins/jquery.colorpicker.js]:
```suggestion
```

File: js/jquery/plugins/jquery.colorpicker.js [hunk 2]
```diff
@@ -41,19 +42,22 @@
...
-    $(document).ready(function() {
-      $.fn.mColorPicker.setTextColor($('.mColorPicker'));
-    });
 
     $(document).on('click', '.mColorPickerTrigger', function () {
-
       $.fn.mColorPicker.colorShow($(this).attr('id').replace('icp_', ''));
     });
 
+    inputs = $();
```
Comment (PierreRambaud) [inline js/jquery/plugins/jquery.colorpicker.js]:
```suggestion
    var inputs = [];
```

File: js/jquery/plugins/jquery.colorpicker.js [hunk 3]
```diff
@@ -41,19 +42,22 @@
...
     this.each(function () {
+      // collect the newly created inputs so that we can update their colors on document ready
+      inputs.push($.fn.mColorPicker.drawPickerTriggers($(this)));
+    });
 
-      $.fn.mColorPicker.drawPickerTriggers($(this));
+    // update the colors of the newly created inputs
+    $(document).ready(function() {
+      $(inputs).each(function(){
```
Comment (PierreRambaud) [inline js/jquery/plugins/jquery.colorpicker.js]:
```suggestion
      $(inputs).each(function() {
```

File: js/jquery/plugins/jquery.colorpicker.js [hunk 4]
```diff
@@ -41,19 +40,22 @@
...
+    var inputs = [];
     this.each(function () {
+      // collect the newly created inputs so that we can update their colors on document ready
+      inputs.push($.fn.mColorPicker.drawPickerTriggers($(this)));
+    });
 
-      $.fn.mColorPicker.drawPickerTriggers($(this));
+    // update the colors of the newly created inputs
+    $(document).ready(function() {
+      inputs.forEach(function(){
```
Comment (PierreRambaud) [inline js/jquery/plugins/jquery.colorpicker.js]:
```suggestion
      inputs.forEach(function() {
```

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @Progi1984 

Thanks for your PR :wink: 
LGTM, QA :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @hibatallahAouadni & @unlocomqx

**Pull request merged**